src/app/util/attribute-storage.h

```diff
@@ -65,8 +65,37 @@
 
 // If we have fixed number of endpoints, then max is the same.
 #ifdef FIXED_ENDPOINT_COUNT
+#ifdef DYNAMIC_ENDPOINT_COUNT
+#define MAX_ENDPOINT_COUNT (FIXED_ENDPOINT_COUNT + DYNAMIC_ENDPOINT_COUNT)
+#else
 #define MAX_ENDPOINT_COUNT FIXED_ENDPOINT_COUNT
 #endif
+#endif
+
+#include <app/common/gen/attribute-type.h>
+
+#define DECLARE_DYNAMIC_ENDPOINT(endpointName, clusterList)                                                                        \
+    EmberAfEndpointType endpointName = { clusterList, sizeof(clusterList) / sizeof(EmberAfCluster), 0 }
+
+#define DECLARE_DYNAMIC_CLUSTER_LIST_BEGIN(clusterListName) EmberAfCluster clusterListName[] = {
```

Comment:
why do we need a define for a variable declaration?

DECLARE_DYNAMIC_CLUSTER_LIST_BEGIN(foo)
  /// stuff
DECLARE_DYNAMIC_CLUSTER_LIST_END;

is much less readable for C++ folks than:

```cpp
EmberAfCluster foo[] {
   /// stuff
};
```

because it hides things in macros.